pkg/virt-handler/node-labeller/cpu_plugin_test.go

```diff
@@ -0,0 +1,147 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package nodelabeller
+
+import (
+	"os"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	kubevirtv1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/pkg/testutils"
+)
+
+var _ = Describe("Node-labeller config", func() {
+	var nlController *NodeLabeller
+	var virtClient *kubecli.MockKubevirtClient
+
+	kv := &kubevirtv1.KubeVirt{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "kubevirt",
+			Namespace: "kubevirt",
+		},
+		Spec: kubevirtv1.KubeVirtSpec{
+			Configuration: kubevirtv1.KubeVirtConfiguration{
+				ObsoleteCPUModels: map[string]bool{
+					"486":        true,
+					"pentium":    true,
+					"pentium2":   true,
+					"pentium3":   true,
+					"pentiumpro": true,
+					"coreduo":    true,
+					"n270":       true,
+					"core2duo":   true,
+					"Conroe":     true,
+					"athlon":     true,
+					"phenom":     true,
+				},
+				MinCPUModel: "Penryn",
+			},
+		},
+	}
+
+	clusterConfig, _, _, _ := testutils.NewFakeClusterConfigUsingKV(kv)
+
+	BeforeSuite(func() {
+		ctrl := gomock.NewController(GinkgoT())
+		virtClient = kubecli.NewMockKubevirtClient(ctrl)
+
+		nlController = &NodeLabeller{
+			namespace:     k8sv1.NamespaceDefault,
+			clientset:     virtClient,
+			clusterConfig: clusterConfig,
+		}
+
+		os.MkdirAll(nodeLabellerVolumePath+"/cpu_map", 0777)
+	})
+
+	AfterSuite(func() {
+		os.Remove(nodeLabellerVolumePath + "/cpu_map")
+	})
+
+	AfterEach(func() {
+		os.Remove(nodeLabellerVolumePath + "/virsh_domcapabilities.xml")
+	})
+
+	It("should return correct cpu file path", func() {
+		fileName := "x86_Penryn.xml"
+		path := getPathCPUFefatures(fileName)
+		Expect(path).To(Equal("/usr/share/libvirt/cpu_map/x86_Penryn.xml"), "cpu file path is not the same")
+	})
+
+	It("should load cpu features", func() {
+		fileName := "x86_Skylake-Server.xml"
```

Comment:
This unit test depends on the existence of external file `/usr/share/libvirt/cpu_map/x86_Skylake-Server.xml`, so it fails if libvirt is not installed.

Can you change the code, so that the directory `/usr/share/libvirt/cpu_map/` is not hard-coded, and this test can load data from a temporary file created by this test or before it? Then it would be also easier to test the case when the file does not exist.